lib/oneview-sdk/resource/api500/c7000/volume.rb

```diff
@@ -151,6 +174,15 @@ def get_volume_template_uri(template_data)
         def generate_snapshot_data(name, description = nil)
           { description: description, name: name }
         end
+
+        # Gets the volume
+        # @raise [OneviewSDK::IncompleteResource] if the name parameter is not set
+        # @return [Array] the array of volumes
+        def find
```

Comment:
The `find` name of this method seems generic, maybe marking that as a private or renaming the method can be better. Or, I guess be good option, you can put the code "to find by name" (It is what this method does) into the `retrieve!` method, it is a simples code and it is used just in this case.

Reply:
This method is already a private method.

Reply:
Not sure what a good name for this could be though, `find_by_name_in_properties` ? Kinda long but...

Reply:
> you can put the code "to find by name" (It is what this method does) into the retrieve! method,

Perhaps following the last comments logic this is actually the best approach, just put it inside `exists?` and since `retrieve!` should be calling `exists?` that'll kill two birds with one stone.